reflection.md

```diff
@@ -2,8 +2,26 @@
 
 ## Which operators were easiest to implement and why?
 
+It was very easy to implement the `||`, `~`, and `<*>` operators since they were
+already basic classes of the `RegularLanguage` trait: `Union`, `Concat`, and
+`Star` respectively.
+So it was effectively just a renaming of these existing classes.
+It was also pretty straightforward to implement `<+>` since it was just a
+combination of two existing classes: `Concat` and `Star`.
+
 ## Which operators were most difficult to implement and why?
 
+The `{n}` operator was definitely the most difficult to write because there were
+a lot of different approaches that could be used.
+I started by using a for loop, but this required defining a variable before the
+loop that is repeatedly modified, so I then tried using a recursive approach.
+The recursive approach seemed much cleaner with the only catch being that I had
+to check for `n == 1` if I wanted to avoid concatenating to `Epsilon` when
+`n > 0`, which wasn't necessary technically, but does make the result cleaner.
+I then tried using `Array.fill` with `reduce` and this eliminated the need for
+the extra `n == 1` check, but the code was more difficult to read, so I opted to
+go with the recursive method instead.
```

Comment:
I agree with your reflection about the operators. I also started with a for loop to illustrate how I understand apply should work, then turned to recursion for a more functional programming approach. I opted out of creating an n = 1 case - I thought it would be ok to concatenate with `Epsilon` given that we have a `simplify` function from last week’s homework.